cmd/flow-aggregator/options.go

```diff
@@ -75,21 +55,14 @@ func newOptions() *Options {
 	}
 }
 
-// addFlags adds flags to fs and binds them to options.
-func (o *Options) addFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&o.configFile, "config", o.configFile, "The path to the configuration file")
-}
-
 // complete completes all the required options.
 func (o *Options) complete(args []string) error {
-	if len(o.configFile) > 0 {
-		c, err := o.loadConfigFromFile(o.configFile)
-		if err != nil {
-			return err
-		}
-		o.config = c
-		o.setDefaults()
+	c, err := o.loadConfigMap()
```

Comment:
I don't think this file is the right place to do this. This is purely to load / validate / complete command-line options. The "run" function would be a better place IMO.

Reply:
Do you mean o.loadConfigMap shouldn't be in option.go? But I didn't change the structure inside option.go. I just changed loadConfigMap() from calling mounted configMap to calling K8s API to access the configMap.

Which "run" function you're referring to?
```
		Run: func(cmd *cobra.Command, args []string) {
			log.InitLogFileLimits(cmd.Flags())
			if err := opts.complete(args); err != nil {
				klog.Fatalf("Failed to complete args: %v", err)
			}
			if err := opts.validate(args); err != nil {
				klog.Fatalf("Failed to validate args: %v", err)
			}
			if err := run(opts); err != nil {
				klog.Fatalf("Error running flow aggregator: %v", err)
			}
		},
```

Reply:
Or do you mean since I remove the --config flag, the whole content in option.go should be there?

Reply:
I meant this function: https://github.com/antrea-io/antrea/blob/2065919367e399256dc5b10db501080e21f8568f/cmd/flow-aggregator/flow-aggregator.go#L79

I think my point is that we should try to make both cases (initial ConfigMap get and ongoing watch) as symmetric as possible, and it feels like most if the code / all of the code dealing with FA configuration should  be moved to the same place in `pkg/flowaggregator`. Instead of providing a config to the initialization function of the FA, maybe the initialization function should be in charge of getting the config from the K8s API.

Reply:
Any updated? If new approach is ok, I'll move these to pkg/flowaggregator.

Reply:
I've moved option.go to pkg/flowaggregator.